icicle-source/src/Icicle/Encoding.hs

```diff
@@ -375,74 +376,73 @@ sourceTypeOfEncoding e
 
 
 
+renderJsonString :: Text -> Text
+renderJsonString =
+  T.decodeUtf8 . BS.toStrict . A.encode . A.String
+
+renderJsonStruct :: Struct -> Text
+renderJsonStruct (Struct kvs0) =
+  let
+    kvs =
+      flip mapMaybe kvs0 $ \(k, v0) -> do
+        v <- renderJsonValue v0
+        pure $
+          renderJsonString (getAttribute k) <> ":" <> v
+  in
+    "{" <> T.intercalate "," kvs <> "}"
+
+renderJsonList :: List -> Text
+renderJsonList (List xs0) =
+  "[" <> T.intercalate "," (mapMaybe renderJsonValue xs0) <> "]"
+
+renderJsonPair :: Value -> Value -> Maybe Text
+renderJsonPair x0 y0 = do
+  x <- renderJsonValue x0
+  y <- renderJsonValue y0
+  pure $
+    "[" <> x <> "," <> y <> "]"
+
+renderJsonValue :: Value -> Maybe Text
+renderJsonValue = \case
+   Tombstone ->
+     Nothing
+   StringValue v ->
+     pure $ renderJsonString v
+   IntValue v ->
+     pure . T.pack $ show v
+   DoubleValue v ->
+     pure . T.decodeUtf8 $ renderDouble v
```

Comment:
We do the JSON rendering ourself here instead of using Aeson, so that we can use `renderDouble` from Anemone